versiondb/tsrocksdb/store.go

```diff
@@ -164,7 +166,9 @@ func (s Store) iteratorAtVersion(storeKey string, start, end []byte, version *in
 	prefix := storePrefix(storeKey)
 	start, end = iterateWithPrefix(prefix, start, end)
 
-	itr := s.db.NewIteratorCF(newTSReadOptions(version), s.cfHandle)
+	readOpts := newTSReadOptions(version)
+	defer readOpts.Destroy()
+	itr := s.db.NewIteratorCF(readOpts, s.cfHandle)
```

Comment:
this one still need the fix in grocksdb right? because `readOpts.Destroy()` will remove ref to the slices which is referenced in iterator.

Reply:
No need hold reference fix if we add destroy since opts reference hold until destroy call, same effect.

Reply:
isn't the iterator lives longer than the destroy call?

Reply:
as long as destroy get call after the last Iter.Next is ok

Reply:
I don't get it, the `Destroy` is called before `iteratorAtVersion` returns, but the iterator itself is used after it returns.

Reply:
I thought you were talking about the test, then might wait for grocksdb change

Reply:
let's patch grocksdb dependency using a fork first?